spec/controllers/users/totp_setup_controller_spec.rb

```diff
@@ -41,64 +41,59 @@
     context 'when user presents invalid code' do
       before do
         sign_in_as_user
+
+        stub_analytics
+        allow(@analytics).to receive(:track_event)
+
         get :new
         patch :confirm, code: 123
       end
 
-      it 'redirects back to authenticator_setup_path' do
+      it 'redirects with an error message' do
         expect(response).to redirect_to(authenticator_setup_path)
-      end
-
-      it 'sets flash[:error] message' do
         expect(flash[:error]).to eq t('errors.invalid_totp')
-      end
-
-      it 'does not enable TOTP for the current user' do
```

Comment:
Thats kind of a shame,  to combine three unit tests into one to speed them up :(

I'm pretty sue there is a variant of `before` we can use the just runs once that will have the same effect as this change on the speed without combining the tests (and therefore loosing some of the test fidelity).